### PR TITLE
VIH-6846 search for multi audio files in Blob

### PR DESCRIPTION
**Pull request by valentinabennett**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-6846

### Change description ###

on Close hearing search by prefix for audio files in Blob

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: VideoAPI/Video.API/Controllers/ConferenceController.cs [hunk 1]
```diff
@@ -491,16 +491,31 @@ private async Task DeleteAudioRecordingApplication(Guid conferenceId)
...
 
-            if (!await azureStorageService.FileExistsAsync(filePath) && conference.ActualStartTime.HasValue)
+            if (!allBlobs.Any() && conference.ActualStartTime.HasValue)
             {
                 var msg = $"Audio recording file not found for hearing: {conference.HearingRefId}";
                 throw new AudioPlatformFileNotFoundException(msg, HttpStatusCode.NotFound);
             }
         }
 
+        private static async Task<IEnumerable<string>> GetAllBlobNamesByFilePathPrefix(string filePathPrefix, IAzureStorageService azureStorageService, string fileExtension = ".mp4")
```
Comment (levibutler) [inline VideoAPI/Video.API/Controllers/ConferenceController.cs:504]: not sure this belongs in a controller?

File: VideoAPI/VideoApi.UnitTests/Controllers/Conference/CloseConferenceTests.cs [hunk 2]
```diff
@@ -56,15 +60,47 @@ public async Task Should_close_conference_and_delete_audio_recording_application
...
             CommandHandlerMock.Verify(c => c.Handle(It.IsAny<CloseConferenceCommand>()), Times.Once);
             AudioPlatformServiceMock.Verify(v => v.DeleteAudioApplicationAsync(It.IsAny<Guid>()), Times.Once);
         }
 
+
+        private async IAsyncEnumerable<BlobClient> GetMockBlobClients()
+        {
+            yield return _blobClientMock.Object;
+
+            await Task.CompletedTask;
```
Comment (levibutler) [inline VideoAPI/VideoApi.UnitTests/Controllers/Conference/CloseConferenceTests.cs:101]: maybe we can do this a little differenly

**Review by levibutler — CHANGES_REQUESTED** [review]

File: VideoAPI/VideoApi.UnitTests/Controllers/Conference/CloseConferenceTests.cs (hunk 2 above)
Comment (levibutler) [reply VideoAPI/VideoApi.UnitTests/Controllers/Conference/CloseConferenceTests.cs:101]: i should have included suggestions..
`IAsyncEnumerable` requires `async` and an async without an await will give a warning so i see why its added.. as its test maybe can ignore..

**Review by rehmanab — APPROVED** [review]

File: VideoAPI/Video.API/Controllers/ConferenceController.cs (hunk 1 above)
Comment (valentinabennett) [reply VideoAPI/Video.API/Controllers/ConferenceController.cs:504]: I think currently I could not move it to service, as actually it's required bigger refactoring in the another parts.   Need to review the implementation of base class and interface for service as there a some duplications for VH and CVP, I think it better to do in separate task. As it just a fix.

**Review by levibutler — APPROVED** [review]
There is still some work to do around refactoring those StorageServices but i think out of scope of this PR. Ill try to get around to it later today
Thanks for moving 👍

**Pull request merged**